### PR TITLE
[NET-3026] filter external events from fargate bpf programs

### PR DESCRIPTION
**Pull request by akarpz**
### What does this PR do?

Testing revealed that eBPF programs in fargate see events from outside the current task. This PR adds a check to fargate programs that returns early if the event pid is not in the same pid namespace as the system probe.

### Motivation

https://datadoghq.atlassian.net/browse/NET-3026

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/ebpf/c/co-re/tracer-fentry.c [hunk 1]
```diff
@@ -105,6 +132,9 @@ static __always_inline int read_conn_tuple_partial_from_flowi6(conn_tuple_t *t,
 
 SEC("fexit/tcp_sendmsg")
 int BPF_PROG(tcp_sendmsg_exit, struct sock *sk, struct msghdr *msg, size_t size, int sent) {
+    if (!event_in_task("fexit/tcp_sendmsg")) {
```
Comment (hmahmood) [inline pkg/network/ebpf/c/co-re/tracer-fentry.c:135]: You could put this behind a macro, e.g. `RETURN_IF_NOT_IN_SYSPROBE_TASK`

File: pkg/network/ebpf/c/tracer.c [hunk 2]
```diff
@@ -9,6 +9,7 @@
 
 #ifdef COMPILE_PREBUILT
 #include "prebuilt/offsets.h"
+#include <uapi/linux/bpf.h>
```
Comment (hmahmood) [inline pkg/network/ebpf/c/tracer.c:12]: Should this be in `tracer-fentry.c`?

File: pkg/network/tracer/connection/fentry/tracer.go [hunk 3]
```diff
@@ -51,6 +53,24 @@ func LoadTracer(config *config.Config, m *manager.Manager, mgrOpts manager.Optio
 		telemetryMapKeys := errtelemetry.BuildTelemetryKeys(m)
 		o.ConstantEditors = append(o.ConstantEditors, telemetryMapKeys...)
 
+		file, err := os.Stat("/proc/self/ns/pid")
+
+		if err != nil {
+			return nil
```
Comment (hmahmood) [inline pkg/network/tracer/connection/fentry/tracer.go:59]: I would return an error here.

File: pkg/network/ebpf/c/tracer.c (hunk 2 above)
Comment (akarpz) [reply pkg/network/ebpf/c/tracer.c]: we actually don't need this at all I don't think

**Review by heyronhay — APPROVED** [review]

**Review by hmahmood — APPROVED** [review]

**Pull request merged**